src/Npgsql/Replication/NpgsqlReplicationConnection.cs

```diff
@@ -0,0 +1,888 @@
+            get => (NpgsqlLogSequenceNumber)unchecked((ulong)Interlocked.Read(ref _lastReceivedLsn));
+            private protected set => Interlocked.Exchange(ref _lastReceivedLsn, unchecked((long)(ulong)value));
+        }
+
+        /// <summary>
+        /// The location of the last WAL byte + 1 flushed to disk in the standby.
+        /// </summary>
+        public NpgsqlLogSequenceNumber LastFlushedLsn
+        {
+            get => (NpgsqlLogSequenceNumber)unchecked((ulong)Interlocked.Read(ref _lastFlushedLsn));
+            set => Interlocked.Exchange(ref _lastFlushedLsn, unchecked((long)(ulong)value));
+        }
+
+        /// <summary>
+        /// The location of the last WAL byte + 1 applied (e. g. written to disk) in the standby.
+        /// </summary>
+        public NpgsqlLogSequenceNumber LastAppliedLsn
+        {
+            get => (NpgsqlLogSequenceNumber)unchecked((ulong)Interlocked.Read(ref _lastAppliedLsn));
+            set => Interlocked.Exchange(ref _lastAppliedLsn, unchecked((long)(ulong)value));
+        }
+
+        /// <summary>
+        /// Send replies at least this often.
+        /// Timeout.<see cref="Timeout.InfiniteTimeSpan"/> disables automated replies.
+        /// </summary>
+        public TimeSpan WalReceiverStatusInterval { get; set; } = TimeSpan.FromSeconds(10d);
+
+        /// <summary>
+        /// Time that receiver waits for communication from master.
+        /// Timeout.<see cref="Timeout.InfiniteTimeSpan"/> disables the timeout.
+        /// </summary>
+        public TimeSpan WalReceiverTimeout
+        {
+            get => _walReceiverTimeout;
+            set
+            {
+                _walReceiverTimeout = value;
+                _requestFeedbackInterval = value == Timeout.InfiniteTimeSpan
+                    ? value
+                    : new TimeSpan(value.Ticks / 2);
+            }
+        }
+
+        private protected abstract ReplicationMode ReplicationMode { get; }
+
+        internal Version PostgreSqlVersion => _npgsqlConnection.PostgreSqlVersion;
+
+        internal NpgsqlConnector Connector => _npgsqlConnection.Connector ?? throw new InvalidOperationException($"The {Connector} property can only be used when there is an active connection");
+
+        ReplicationConnectionState State
+        {
+            get => (ReplicationConnectionState)_state;
+            set
+            {
+                // Disposed and Broken are final states. We never leave them.
+                if (State == ReplicationConnectionState.Disposed || State == ReplicationConnectionState.Broken)
+                    return;
+
+                _state = (int)value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the wait time before terminating the attempt  to execute a command and generating an error.
+        /// </summary>
+        /// <value>The time to wait for the command to execute. The default value is 30 seconds.</value>
+        public TimeSpan CommandTimeout
```

Comment:
I think `ReplicationTimeout` would be a bit confusing because this timeout is used when executing commands on the replication connection and not while replicating (streaming).

Reply:
Good point. But from the user's perspective there aren't any commands either (in the sense of NpgsqlCommand).

Should we go for `Timeout` then?

Reply:
> Should we go for Timeout then?

👍

Reply:
On second look the existing CommandTimeout seems better - it may indicate more clearly that the timeout doesn't cover actual replication, only command execution. It's not a very good name, but Timeout is ambiguous in a problematic way.